client/src/components/bash-button.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import { Button, Typography } from "@material-ui/core";
+import React from "react";
+import { navigate } from "gatsby";
+import PropTypes from "prop-types";
+import { useWithReviewAnswerState } from "hooks/graphql/use-with-review-answer-state";
+import { Answer, MentorType, Status, UtteranceName } from "types";
+
+export default function BashButton(props: {
+  accessToken: string;
+  setThumbnail: (file: File) => void;
+}): JSX.Element {
+  const { mentor } = useWithReviewAnswerState(props.accessToken, {
+    subject: undefined,
+  });
+  const bash = {
+    text: "",
+    reason: "",
+    action: () => {
+      undefined;
+    },
+  };
+  const idle = mentor?.answers.find(
+    (a) => a.question.name === UtteranceName.IDLE
+  );
+  const categories: {
+    subjectName: string;
+    subject: string;
+    categoryName: string;
+    category: string | undefined;
+    answers: Answer[];
+  }[] = [];
+
+  mentor?.subjects.forEach((s) => {
+    categories.push({
+      subjectName: s.name,
+      subject: s._id,
+      categoryName: "Uncategorized",
+      category: undefined,
+      answers: mentor?.answers.filter((a) =>
+        s.questions
+          .filter((q) => !q.category)
+          .map((q) => q.question._id)
+          .includes(a.question._id)
+      ),
+    });
+    s.categories.forEach((c) => {
+      categories.push({
+        subjectName: s.name,
+        subject: s._id,
+        categoryName: c.name,
+        category: c.id,
+        answers: mentor?.answers.filter((a) =>
+          s.questions
+            .filter((q) => q.category?.id === c.id)
+            .map((q) => q.question._id)
+            .includes(a.question._id)
+        ),
+      });
+    });
+  });
+  const firstIncomplete = categories.find((c) =>
+    c.answers.find((a) => a.status === Status.INCOMPLETE)
+  );
+
+  switch (true) {
```

Comment:
Instead of `switch` would be cleaner and easier to understand if you break this out into a function. 

```typescript
interface RecommendedAction {
  text: string;
  reason: string;
  action: () => void
}
function recommend(): RecommendedAction {
  if (!mentor.thumbnail) {
    return {
      text: "Add a thumbnail",
      // ...
    }
  }
}
```